### PR TITLE
comment threading proof of concept

### PR DESCRIPTION
**Pull request by anselmdk**
See #28.

### CODE REVIEW

**Comment by tractorcow** [comment]
I see a problem with this approach if you wanted to gracefully degrade in environments where javascript is disabled. At the moment the current comments module should work without it, but additional scripts allow functionality to be enhanced in case it is enabled. :)

File: code/controllers/CommentingController.php [hunk 1]
```diff
@@ -207,6 +208,36 @@ public function approve() {
...
+	 * 
+	 * @return string
+	 */
+	public function replyform(){
+		$p = $this->getURLParams();
+		$cID = (int) $p['ID'];
+		$c = Comment::get()->ByID($cID);
+
+		$controller = new CommentingController();		
+		$controller->setOwnerRecord($c);
```
Comment (dhensby) [inline code/controllers/CommentingController.php]: The controller needs the `init()` function running

File: code/controllers/CommentingController.php [hunk 2]
```diff
@@ -207,6 +208,36 @@ public function approve() {
...
+		$controller->setBaseClass('Comment');
+		$controller->setOwnerController(Controller::curr());
+		
+		$form = $controller->CommentsForm();
+		
+		//setting return url
+		if (isset($_GET['ReturnURL'])) {
```
Comment (dhensby) [inline code/controllers/CommentingController.php]: Don't manipulate superglobals. use `$this->request->getVar('ReturnURL');`

File: code/controllers/CommentingController.php [hunk 3]
```diff
@@ -207,6 +208,36 @@ public function approve() {
...
+			$form->loadDataFrom(array(
+				'ReturnURL'		=> $_GET['ReturnURL'],
+			));
+		}
+		
+		return $form->forTemplate();
```
Comment (dhensby) [inline code/controllers/CommentingController.php]: You don't need to explicitly render the form. If you don't then it makes extending the form easier.

**Comment by dhensby** [comment]
I think this could fairly easily be made to work without JS too

**Comment by tractorcow** [comment]
I'm going to close this for now, since we've added nested comments in the master branch.

Thanks @anselmdk for your hard work and suggestions. :)

**Comment by anselmdk** [comment]
@tractorcow np - happy it's finally been implemented!

**Pull request closed without merging**